src/box/lua/net_box.lua

```diff
@@ -59,11 +60,23 @@ local IPROTO_FEATURE_NAMES = {
     [4]     = 'pagination',
 }
 
+local check_select_features = function(opts, remote)
+    if opts == nil or opts.after == nil and not opts.fetch_pos then
+        return
+    elseif not remote.peer_protocol_features.pagination then
+        return error("Pagination is not supported by remote server")
```

Comment:
Nit: `box.error` with the `ER_UNSUPPORTED` code looks more appropriate here.

Reply:
Done